ca/transport.go

```diff
@@ -135,6 +136,16 @@ func (c *MutableTLSCreds) LoadNewTLSConfig(newConfig *tls.Config) error {
 	return nil
 }
 
+// UpdateCAs updates the root CAs and client CAs of the existing TLS config in place
+func (c *MutableTLSCreds) UpdateCAs(rootCAs, clientCAs *x509.CertPool) {
+	c.Lock()
+	defer c.Unlock()
+	config := tlsconfig.Clone(c.config)
```

Comment:
I tried that originally, but it's not really safe to modify a `tls.Config` while it's being used I think.  I ended up with a data race in the tests, since a `tls.Config` stores some handshake/session state.  The comment for `tls.Config` contains:

`After one has been passed to a TLS function it must not be modified.`